keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -85,11 +88,15 @@ open class SqlResourceRepository(
    * Constructs a resource object from its database representation
    */
   private fun constructResource(kind: String, metadata: String, spec: String) =
-    Resource(
-      parseKind(kind),
-      objectMapper.readValue<Map<String, Any?>>(metadata).asResourceMetadata(),
-      objectMapper.readValue(spec, resourceTypeIdentifier.identify(parseKind(kind)))
-    )
+    specMigrators
+      .migrate(parseKind(kind), objectMapper.readValue(spec))
+      .let { (endKind, endSpec) ->
+        Resource(
+          endKind,
+          objectMapper.readValue<Map<String, Any?>>(metadata).asResourceMetadata(),
+          objectMapper.convertValue(endSpec, resourceTypeIdentifier.identify(endKind))
+        )
+      }
```

Comment:
Inside of `SqlResourceRepository` this function turns the row results into resources with the correct spec type. Now I'm just using the migrators up front so that we only ever try to parse the latest type of spec. We don't need to keep the spec classes for the older versions around.